src/main/java/org/folio/consortia/service/impl/SharingRoleCapabilitySetService.java

```diff
@@ -58,7 +58,16 @@ protected Object getPayload(SharingRoleCapabilitySetRequest sharingRoleCapabilit
 
   @Override
   protected String getPayloadId(ObjectNode payload) {
-    return payload.get("roleId").asText();
+    return payload.get(ID_FIELD).asText();
+  }
+
+  @Override
+  protected String getUrl(SharingRoleCapabilitySetRequest request, HttpMethod httpMethod) {
+    String url = request.getUrl();
+    if (httpMethod.equals(HttpMethod.PUT) || httpMethod.equals(HttpMethod.DELETE)) {
```

Comment:
this if condition is the same for all subclasses so can be moved to parent class